cmd/ops_agent_windows/install_windows.go

```diff
@@ -60,8 +64,60 @@ func install() error {
 	}
 	// Registering with the event log is required to suppress the "The description for Event ID 1 from source Google Cloud Ops Agent cannot be found" message in the logs.
 	if err := eventlog.InstallAsEventCreate(serviceName, eventlog.Error|eventlog.Warning|eventlog.Info); err != nil {
-		// Ignore error since it likely means the event log already existss.
+		// Ignore error since it likely means the event log already exists.
 	}
-	// Automatically start the Ops Agent service
+	// Automatically start the Ops Agent service.
 	return handles[0].Start()
 }
+
+func uninstall() error {
+	m, err := mgr.Connect()
+	if err != nil {
+		return err
+	}
+	defer m.Disconnect()
+	errs := []string{}
+	last := len(services) - 1
+	for i := range services {
+		// Have to remove the services in reverse order.
+		s := services[last-i]
+		serviceHandle, err := m.OpenService(s.name)
+		if err != nil {
+			// Service does not exist, so nothing to delete.
+			continue
+		}
+		defer serviceHandle.Close()
+		if err := stopService(serviceHandle, 30 * time.Second); err != nil {
+			// Don't return until all services have been processed.
+			errs = append(errs, err.Error())
+		}
+		if err := serviceHandle.Delete(); err != nil {
+			// Don't return until all services have been processed.
+			errs = append(errs, err.Error())
+		}
+	}
+	if len(errs) != 0 {
+		return fmt.Errorf(strings.Join(errs, "\n"))
+	}
+	return nil
+}
+
+func stopService(serviceHandle *mgr.Service, timeout time.Duration) error {
+	// TODO: is it possible to use channel-based timeout handling here?
```

Comment:
I don't think there's much point unless you have the option to use a blocking or channel-based API for listening for the stop event.